src/main/java/com/dataloom/datastore/services/SearchService.java

```diff
@@ -520,4 +491,19 @@ private NeighborEntityDetails getNeighborEntityDetails(
         return null;
     }
 
+    private static List<SetMultimap<Object, Object>> getResults(
+            EntityDatastore dataManager,
+            SearchResult result,
+            Map<UUID, PropertyType> authorizedPropertyTypes ) {
+        result.getHits().parallelStream()
+                .map( hit -> (String) hit.get( "id" ) )
+                .map( UUID::fromString )
+                .collect( Collectors.toSet() );
+        Set<UUID> ids = result.getHits().parallelStream()
+                .map( hit -> (String) hit.get( "id" ) )
+                .map( UUID::fromString )
+                .collect( Collectors.toSet() );
+        return dataManager.getEntities( ids, authorizedPropertyTypes ).collect( Collectors.toList() );
```

Comment:
but yes that is a bit hacky, we should just make every entity have an implicit property called id with reserved FQN.